### PR TITLE
Show clear warning for experimental features

### PR DESCRIPTION
**Pull request by maximedenes**
I tried to make the way some features are marked as experimental more uniform in the manual.

Maybe this is also a good opportunity to ask what "experimental" means. Looking at what is considered experimental today, it's not easy to recognize a pattern.

### CODE REVIEW

File: doc/sphinx/addendum/micromega.rst [hunk 1]
```diff
@@ -194,10 +194,11 @@ a proof.
 .. tacn:: nra
    :name: nra
 
-   This tactic is an *experimental* proof procedure for non-linear
-   arithmetic. The tactic performs a limited amount of non-linear
-   reasoning before running the linear prover of :tacn:`lra`. This pre-processing
-   does the following:
+   The status of this tactic is experimental.
```
Comment (silene) [inline doc/sphinx/addendum/micromega.rst]: You forgot `.. warning::` there. That said, I would not mark this tactic as experimental. It is not like its behavior might suddenly change in the next version.

File: doc/sphinx/language/gallina-extensions.rst [hunk 2]
```diff
@@ -592,10 +592,12 @@ This example emphasizes what the printing options offer.
 Advanced recursive functions
 ----------------------------
 
-The following experimental command is available when the ``FunInd`` library has been loaded via ``Require Import FunInd``:
+The following command is available when the ``FunInd`` library has been loaded via ``Require Import FunInd``:
 
 .. cmd:: Function @ident {* @binder} { @decrease_annot } : @type := @term
 
+   .. warning:: The status of this command is experimental.
```
Comment (silene) [inline doc/sphinx/language/gallina-extensions.rst]: As with `nra`, I don't think `Function` will suddenly change.

File: doc/sphinx/proof-engine/tactics.rst [hunk 3]
```diff
@@ -1992,17 +1992,20 @@ analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`)
...
-   :cite:`DBLP:conf/types/McBride00` and the work of Cristina Cornes around
-   inversion :cite:`DBLP:conf/types/CornesT95`. From an instantiated
-   inductive predicate and a goal, it generates an equivalent goal where
-   the hypothesis has been generalized over its indexes which are then
-   constrained by equalities to be the right instances. This permits to
-   state lemmas without resorting to manually adding these equalities and
-   still get enough information in the proofs.
+   .. warning::
+
+      The status of this tactic is experimental.
```
Comment (silene) [inline doc/sphinx/proof-engine/tactics.rst]: Same here. I don't think anyone is working on it.

File: doc/sphinx/proof-engine/tactics.rst [hunk 4]
```diff
@@ -4212,10 +4215,14 @@ some incompatibilities.
 .. tacn:: firstorder
    :name: firstorder
 
-   The tactic :tacn:`firstorder` is an experimental extension of :tacn:`tauto` to
-   first- order reasoning, written by Pierre Corbineau. It is not restricted to
-   usual logical connectives but instead may reason about any first-order class
-   inductive definition.
+   .. warning::
+
+      The status of this tactic is experimental.
```
Comment (silene) [inline doc/sphinx/proof-engine/tactics.rst]: Same here.

File: doc/sphinx/user-extensions/proof-schemes.rst [hunk 5]
```diff
@@ -198,14 +198,17 @@ Generation of induction principles with ``Functional`` ``Scheme``
...
-   generating automatically induction principles corresponding to
-   (possibly mutually recursive) functions. First, it must be made
-   available via ``Require Import FunInd``.
-   Each :n:`@ident__i` is a different mutually defined function
-   name (the names must be in the same order as when they were defined). This
-   command generates the induction principle for each :n:`@ident__i`, following
-   the recursive structure and case analyses of the corresponding function
+   .. warning::
+
+      The status of this command is experimental.
```
Comment (silene) [inline doc/sphinx/user-extensions/proof-schemes.rst]: Same here.

File: doc/sphinx/addendum/universe-polymorphism.rst [hunk 6]
```diff
@@ -279,6 +279,8 @@ Cumulativity Weak Constraints
 
 .. flag:: Cumulativity Weak Constraints
 
+   .. warning:: The status of this flag is experimental
```
Comment (SkySkimmer) [inline doc/sphinx/addendum/universe-polymorphism.rst]: Here experimental means it's an unstable feature
Admittedly I think we haven't made much effort in stabilizing the heuristics of universe constraints in general so specifically warning for this one heuristic may be questionable

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 7]
```diff
@@ -986,13 +986,17 @@ Controlling display
...
-   This option configures the display of warnings. It is experimental, and
-   expects, between quotes, a comma-separated list of warning names or
-   categories. Adding - in front of a warning or category disables it, adding +
-   makes it an error. It is possible to use the special categories all and
-   default, the latter containing the warnings enabled by default. The flags are
-   interpreted from left to right, so in case of an overlap, the flags on the
-   right have higher priority, meaning that `A,-A` is equivalent to `-A`.
+   .. warning::
+
+      The status of this option is experimental.
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/vernacular-commands.rst]: If experimental means some users should be discouraged from using it (and with a warning, this is the kind of meaning it carries), then I would vote against this status for the `Warnings` command: I sure would like to encourage user to finely tune warnings, and try to build with warnings as errors. Sure, we'd like to refine the API but if we do, we will still be backward compatible.

**Review by Zimmi48 — COMMENTED** [review]
Overall, I'd say that Ltac2, Dune, and the mangle-name feature are the only ones deserving an experimental warning, especially given that the attribute feature never had such a warning.

**Comment by maximedenes** [comment]
@coq/doc-maintainers anybody wants to take over this PR or should I erase the branch?

**Pull request closed without merging**